llvm/lib/Transforms/Vectorize/VectorCombine.cpp

```diff
@@ -29,6 +29,7 @@
 #include "llvm/IR/PatternMatch.h"
 #include "llvm/Support/CommandLine.h"
 #include "llvm/Transforms/Utils/Local.h"
+#include "llvm/Transforms/Utils/LoopUtils.h"
```

Comment:
Is this required? I don't find anything related to LoopUtils.

Reply:
getArithmeticReductionInstruction() is defined there.

Reply:
Yes, the reduction intrinsics mapping helpers are all defined in LoopUtils.h